src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2321,6 +2321,46 @@ public static void TestNameWithInvariantCulture()
 
         }
 
+        [Theory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [InlineData("America/Buenos_Aires", "America/Argentina/Buenos_Aires")]
+        [InlineData("America/Catamarca", "America/Argentina/Catamarca")]
+        [InlineData("America/Cordoba", "America/Argentina/Cordoba")]
+        [InlineData("America/Jujuy", "America/Argentina/Jujuy")]
+        [InlineData("America/Mendoza", "America/Argentina/Mendoza")]
+        [InlineData("America/Indianapolis", "America/Indiana/Indianapolis")]
+        public static void TestTimeZoneIdBackwardCompatibility(string oldId, string currentId)
+        {
+            TimeZoneInfo oldtz = TimeZoneInfo.FindSystemTimeZoneById(oldId);
+            TimeZoneInfo currenttz = TimeZoneInfo.FindSystemTimeZoneById(currentId);
+
+            Assert.Equal(oldtz.StandardName, currenttz.StandardName);
+            Assert.Equal(oldtz.DisplayName, currenttz.DisplayName);
+        }
+
+        [Theory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [InlineData("America/Buenos_Aires")]
+        [InlineData("America/Catamarca")]
+        [InlineData("America/Cordoba")]
+        [InlineData("America/Jujuy")]
+        [InlineData("America/Mendoza")]
+        [InlineData("America/Indianapolis")]
+        public static void ChangeLocalTimeZone(string id) 
+        {
+            TimeZoneInfo.ClearCachedData();
+            Environment.SetEnvironmentVariable("TZ", id);
+
+            TimeZoneInfo localtz = TimeZoneInfo.Local;
+            TimeZoneInfo tz = TimeZoneInfo.FindSystemTimeZoneById(id);
+
+            Assert.Equal(tz.StandardName, localtz.StandardName);
+            Assert.Equal(tz.DisplayName, localtz.DisplayName); 
+
+            TimeZoneInfo.ClearCachedData();
+            Environment.SetEnvironmentVariable("TZ", "");
```

Comment:
This should be in a `finally{}` block, so that we ensure that it is always cleared, and TZ is reset.

Reply:
Strictly speaking it should probably save the original TZ value and restore it in the finally block.  We run the risk of breaking some other test otherwise right?